### PR TITLE
Use GraphQL endpoint for retrieving lite cocina objects

### PR DESCRIPTION
**Pull request by justinlittman**
# Why was this change made? 🤔
Performance

# How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (including writing to shared file systems or interacting with other SDR APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

⚡ ⚠ If this change updates the Argo UI, run all integration tests that use Argo and create a PR on the integration test repo to fix anything this change breaks. ⚡

# Does your change introduce accessibility violations? 🩺

⚡ ⚠ Please ensure this change does not introduce accessibility violations (at the WCAG A or AA conformance levels); if it does, include a rationale. See the [Infrastructure accessibility guide](https://github.com/sul-dlss/DeveloperPlaybook/blob/main/best-practices/infra-accessibility.md) for more detail. ⚡

### CODE REVIEW

File: spec/features/item_registration_spec.rb [hunk 1]
```diff
@@ -122,6 +122,7 @@
       fill_in "Label", with: "object title"
 
       click_button "Register"
+      save_screenshot("test.png")
```
Comment (ndushay) [inline spec/features/item_registration_spec.rb:125]: cruft?

**Review by ndushay — APPROVED** [review]
approving module some comments

File: app/controllers/catalog_controller.rb [hunk 2]
```diff
@@ -237,17 +237,17 @@ def show
...
     authorize! :read, @cocina
 
     @workflows = WorkflowService.workflows_for(druid: params[:id])
-
     milestones = MilestoneService.milestones_for(druid: params[:id])
     object_client = Dor::Services::Client.object(params[:id])
+    versions = object_client.version.inventory
     @milestones_presenter = MilestonesPresenter.new(milestones:,
-      versions: object_client.version.inventory)
+      versions: versions)
```
Comment (ndushay) [inline app/controllers/catalog_controller.rb:250]: it's stupid to have this on a second line, IMO.

Comment (justinlittman) [reply app/controllers/catalog_controller.rb:250]: Fixed.

File: spec/features/item_registration_spec.rb (hunk 1 above)
Comment (justinlittman) [reply spec/features/item_registration_spec.rb:125]: Yes. Removed.

**Pull request merged**